src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * 
+ */
+package seedu.addressbook.data.person;
+
+/**
+ * @author Nat
+ *
+ */
+public class Contact {
+	
+	  protected String value;
+	  protected boolean isPrivate;
+	  
+	  @Override
+	    public String toString() {
+	        return value;
+	    }
+
+	    @Override
+	    public boolean equals(Object other) {
+	        return other == this // short circuit if same object
+	                || (other instanceof Address // instanceof handles nulls
```

Comment:
comparing with an instandof Address is WRONG. Should be an instance of Contact.

If you are overriding the equal methods in the subclass to check if its a instance of each of those respective types, make sure you override the hashCode() as well.